Core/automation/lib/python/core/utils.py

```diff
@@ -315,3 +316,61 @@ def postUpdate(item_or_item_name, new_value):
     events.postUpdate(item, new_value)
+
+def get_item_state(item_or_item_name, default=None, return_type=None, unit=None):
+    """
+    Gets an Item's state, optionally converting it to a specified type and/or unit 
+    for Quantity Types 
+
+    Args:
+        item_or_item_name (GenericItem or str): Item or name of the Item
+        default (Any): value to return if Item does not exist, if Item's state is ``NULL``
+            or ``UNDEF`` or if attempts to convert Item's state fail.
+        return_type (Any): Python or Java data type to attempt to return the Item's state
+            as, if not provided or `None` no conversion will be done. If the conversion
+            fails the default will be returned 
+        unit (str): if Item's state is a QuantityType, attempt to convert to this unit. This 
+            is done before any conversion attempts if ``return_type`` is provided.
+
+    Returns:
+        Item's state if Item exists and its state is not ``NULL`` or ``UNDEF`` if no
+        ``return_type`` or ``unit`` is provided, otherwise ``default`` will be returned.
+        If ``return_type`` is provided the Item's state will be returned as that type if
+        possible, otherwise ``default`` will be returned.
+        If ``unit`` is provided and the Item's state is a QuantityType it will be converted
+        to those units before attempting to convert to ``return_type``, if the unit conversion
+        fails ``defualt`` will be returned.
+    """
+    
+    type_methods = {int: "intValue", float: "floatValue", str: "toString"}
+    conv_methods = {
+        datetime: to_python_datetime,
+        ZonedDateTime: to_java_zoneddatetime,
+        JodaDateTime: to_joda_datetime,
+        JavaCalendar: to_java_calendar
+    }
+    item = validate_item(item_or_item_name) # type: t.Union[GenericItem, None]
+    if item:
+        state = item.getState() # type: State
+        if isinstance(state, QuantityType) and unit is not None:
+            try:
+                state = state.toUnit(unit)
+            except:
+                LOG.warn(u"Item '{}' state '{}' cannot be converted to '{}' units".format(item.name, item.state, unit))
+                return default
+        if not isinstance(state, UnDefType):
+            if return_type is None:
+                return state
+            elif return_type in type_methods:
+                if hasattr(state, type_methods[return_type]):
+                    state = getattr(state, type_methods[return_type])()
+            elif return_type in conv_methods:
+                return_type = conv_methods[return_type]
+            try: # attempt to cast
+                return return_type(state)
+            except:
+                LOG.warn(u"Item '{}' type '{}', cannot be converted to '{}'".format(item.name, type(state), return_type))
+                pass
+    else:
+        LOG.warn(u"Item does not exist or is uninitialized, returning default value")
```

Comment:
`validate_item` will give us a warning if the item doesn't exist. Giving a warning if the item is uninitialized is redundant, this is the expected behaviour of the function and the caller should be handling the return of the default value.

Reply:
Removed